template.tpl

```diff
@@ -794,46 +832,51 @@ const unixMillisToISO = (unixMillis) => {
   return formatISO(year, month, day, hour, minute, second, millis);
 };
 
-/*
+/**
  * Given an array and a configuration object,
  *  returns the element from a single element array or the array itself.
  *
- * @param arr {Array} - the input array
- * @param tagConfig {Object} - the tag configuration object
- *        tagConfig.extractFromArray - whether to extract a single element
- * @returns - the array or its single element
+ * @param {Array} arr - The input array
+ * @param {Object} tagConfig - The tag configuration object
+ * @param {boolean} tagConfig.extractFromArray - Whether to extract a single element
+ * @returns {*} The array or its single element
  */
 const extractFromArrayIfSingleElement = (arr, tagConfig) =>
   arr.length === 1 && tagConfig.extractFromArray ? arr[0] : arr;
 
-/*
+/**
  * Cleans a name from the GTM-SS Snowplow prefix ('x-sp-').
  *
- * @param prop {string} - the property name
- * @returns - the property name with the GTM-SS Snowplow prefix removed.
+ * @param {string} prop - The property name
+ * @returns {string} The property name with the GTM-SS Snowplow prefix removed.
  */
 const cleanPropertyName = (prop) => prop.replace('x-sp-', '');
 
-/*
+/**
  * Parses a Snowplow schema to the expected major version format,
  *  also prefixed so as to match the contexts' output of the Snowplow Client.
  *
- * @param schema {string} - the input schema
- * @returns - the expected output client event property
+ * @param schema {string} - The input schema
+ * @returns {string} The expected output client event property
  */
 const parseSchemaToMajorKeyValue = (schema) => {
   if (schema.indexOf('x-sp-contexts_') === 0) return schema;
   if (schema.indexOf('contexts_') === 0) return 'x-sp-' + schema;
   if (schema.indexOf('iglu:') === 0) {
-    let fixed = replaceAll(
-      replaceAll(
-        schema.replace('iglu:', '').replace('jsonschema/', ''),
-        '.',
-        '_'
-      ),
-      '/',
-      '_'
-    );
+    const rexp = createRegex('[./]', 'g');
+    let fixed = schema
+      .replace('iglu:', '')
+      .replace('jsonschema/', '')
+      .replace(rexp, (match) => {
```

Comment:
Can't this just be `.replace(rexp, '_')`?

Reply:
Much better that way, thank you for spotting! Changed.